plugins/bigquery/dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -15,51 +15,67 @@
 {% endmacro %}
 
 
-{% macro bq_partition_merge(strategy, tmp_relation, target_relation, sql, unique_key, partition_by, dest_columns) %}
+{% macro bq_insert_overwrite(tmp_relation, target_relation, sql, unique_key, partition_by, partitions, dest_columns) %}
   {%- set partition_type =
       'date' if partition_by.data_type in ('timestamp, datetime') 
       else partition_by.data_type -%}
 
-  {% set predicate -%}
-      {{ partition_by.render(alias='DBT_INTERNAL_DEST') }} in unnest(dbt_partitions_for_upsert)
-  {%- endset %}
-
-  {%- set source_sql -%}
-  (
-    select * from {{ tmp_relation }}
-  )
-  {%- endset -%}
-
-  -- generated script to merge partitions into {{ target_relation }}
-  declare dbt_partitions_for_upsert array<{{ partition_type }}>;
-  declare _dbt_max_partition {{ partition_by.data_type }};
-
-  set _dbt_max_partition = (
-      select max({{ partition_by.field }}) from {{ this }}
-  );
-
-  -- 1. create a temp table
-  {{ create_table_as(True, tmp_relation, sql) }}
-
-  -- 2. define partitions to update
-  set (dbt_partitions_for_upsert) = (
-      select as struct
-          array_agg(distinct {{ partition_by.render() }})
-      from {{ tmp_relation }}
-  );
+  {% if partitions is not none and partitions != [] %} {# static #}
+  
+      {% set predicate -%}
+          {{ partition_by.render(alias='DBT_INTERNAL_DEST') }} in (
+              {% for partition in partitions %}
+                '{{partition}}' {{- ',' if not loop.last -}}
```

Comment:
I'm not sure that we want to quote here -- seems like something that should be left to the user? If we stringify, then we can't support:
 - expressions
 - integer buckets

Reply:
we can provide tools to help here for sure

Reply:
Also: can you confirm that `where ... in ( .... )` prunes partitions the same as 
```
where 
  field = ... or
  field = ... or 
  ...
```

Reply:
Great point. I was thinking about whether we quote or not based on the `partition-by.data_type`, but you're right that a user may want to pass a SQL expression.

I've confirmed that `where field in ( .... )` and `where field = ... or field = ...` are pruned the same, and that both allow the user to pass SQL expressions (e.g. `date_sub(current_date, interval 1 day)`) in addition to string literals.

By "left to the user," you mean that they would supply the config as
```sql
{{config(
    materialized = 'incremental',
    unique_key = 'id',
    partition_by = {'field': 'date_hour', 'data_type': 'timestamp'},
    incremental_strategy = 'insert_overwrite',
    partitions = ["'2020-03-08'", "'2020-03-09'", "'2020-03-10'"]
)}}
```

This is an ok answer tbh. What sort of tooling do you imagine would make this easier?

Reply:
Confirmed that we do not want to quote the results. In the future, it would be good if we could give some syntactic sugar for using Jinja/datetime macros, but it also occurs that the following would work for "always run last 3 days":
```sql
{{config(
    materialized = 'incremental',
    unique_key = 'id',
    partition_by = {'field': 'date_hour', 'data_type': 'timestamp'},
    incremental_strategy = 'insert_overwrite',
    partitions = ['current_date', 'date_sub(current_date, interval 1 day)', 'date_sub(current_date, interval 2 day)']
)}}
```